### DISCUSSION TITLE
ios build is failing after upgrading to xcode 14.3 and instance type mac_mini_m1

### DISCUSSION THREAD

**Discussion by pranaynemavrize** [codemagic-ci-cd/codemagic-docs#1587] [Q&A] [answered]
Hello there,

Android is working fine but ios build is failing to archive without any error shown.

After upgrading to xcode 14.3 and instance type mac_mini_m1, i am not able to build ios from codemagic while it is working fine on local builds.

Failed on command - 

cd /Users/builder/clone/ios
xcode-project build-ipa --workspace TSC.xcworkspace --scheme TSC --export-flags='-allowProvisioningUpdates'

build id - 64422aafb95109f523089a59

<img width="700" alt="Screenshot 2023-04-21 at 3 50 13 PM" src="https://user-images.githubusercontent.com/111505413/233613045-e69d2f62-7637-4f63-b7cc-62ec2deb8557.png">

**Comment by icarusdust** [comment, marked as answer]
Hi @pranaynemavrize, you are getting:

`error: Provisioning profile "XXXXXXXXX" doesn't include the currently selected device "builderâ€™s Virtual Machine" (identifier 0000FE00-343A79319AA17942). (in target 'ImageNotification' from project 'TSC')`

Add  ` --archive-flags="-destination 'generic/platform=iOS'"` to `xcode-project build-ipa`. So, it will look like:

`xcode-project build-ipa --archive-flags="-destination 'generic/platform=iOS'"`

**Reply by icarusdust** [reply]
Besides, to get more of Xcode logs, you can add `- /tmp/xcodebuild_logs/*.log` to the artefacts section which will generate a zip file containing all the logs.

**Reply by pranaynemavrize** [reply]
@icarusdust Thanks for your answer i will try and let you know .

**Reply by pranaynemavrize** [reply]
@icarusdust  Thank you so much its working now
